### PR TITLE
Add overview and more detailed documentation of tools/services

### PR DESCRIPTION
**Pull request by ES-Alexander**
- [x] add overview
- [x] review installation instructions
- [x] convert 'configuration' page into 'getting started' info (review w.r.t. new welcome wizard)
- [x] add advanced usage and configuration info (aesthetics, pirate mode, page for each service, etc)

### CODE REVIEW

**Comment by ES-Alexander** [comment]
#28 should be merged first

File: advanced-usage/index.md [hunk 1]
```diff
@@ -0,0 +1,372 @@
...
+{{ simple_pirate_image(src="sidebar", width=200) }}
+
+##### Settings
+
+{{ simple_pirate_image(src="settings", width=300) }}
+
+- Reset BlueOS settings
+   - Remove existing camera/endpoint/bridges configuration
+- Pirate mode toggle
+   - Access or hide advanced functionality
```
Comment (patrickelectric) [inline advanced-usage/index.md:89]: (Advanced users only, this mode is not recommended for normal use)

File: advanced-usage/index.md [hunk 2]
```diff
@@ -0,0 +1,372 @@
...
+   - Recommended before turning off vehicle power
+- Reboot
+   - Reboot onboard computer
+{% simple() %}
+- Soft restart
+   - Do a software-restart of BlueOS (without restarting the computer)
+   - Generally sufficient for most 'reboot' requirements
+{% end %}
+{% pirate() %}
+- Restart core container
```
Comment (patrickelectric) [inline advanced-usage/index.md:108]: - Restart code container (the same as Soft restart)

File: advanced-usage/index.md [hunk 3]
```diff
@@ -0,0 +1,372 @@
...
+
+##### Feedback
+
+{{ easy_image(src="feedback", width=300) }}
+
+Submit feedback about BlueOS via:
+- Issues on the GitHub repository (allows easily tracking changes, 
+and notification when complete/fixed)
+- Posts on the Blue Robotics forum (allows easy discussion with the community)
+- Messages in the Blue Robotics Slack (for internal use)
```
Comment (patrickelectric) [inline advanced-usage/index.md:121]: Slack is only available in pirate mode

File: advanced-usage/index.md [hunk 4]
```diff
@@ -0,0 +1,372 @@
...
+
+## Vehicle
+
+### General
+The "General" page provides basic info about the active autopilot, along with
+options to:
+- Change board (select a connected board, or run an
```
Comment (patrickelectric) [inline advanced-usage/index.md:128]: This option is only available in pirate mode

File: advanced-usage/index.md [hunk 5]
```diff
@@ -0,0 +1,372 @@
...
+
+{{ simple_pirate_image(src="general", width=600) }}
+
+### Firmware
+- [ArduPilot](ardupilot.org) family of firmwares only
+- Choose firmware to install
+   - Select from the online repository
+      - Select vehicle type (Sub / Rover / Plane / Copter)
+      - Select desired release and stability level
+         - Official - The latest stable release. Recommended for most users.
```
Comment (patrickelectric) [inline advanced-usage/index.md:144]: This is not available anymore on the combobox

File: advanced-usage/index.md [hunk 6]
```diff
@@ -0,0 +1,372 @@
...
+         - Stable - A production-ready release. Suitable for most users.
```
Comment (patrickelectric) [inline advanced-usage/index.md:145]: (E.g: Stable-4.0.3)

File: advanced-usage/index.md [hunk 7]
```diff
@@ -0,0 +1,372 @@
...
+- BlueOS automatically detects H264-encoded video streams
+- The first time BlueOS starts up it will auto-configure any cameras that are 
+connected at that time, with UDP streams counting up from port 5600
+   - e.g. a second camera at first startup would be streamed to port 5601
+   - Auto-configuration also occurs if the [settings are reset](#settings)
+{% pirate() %}
+      - It's also possible to manually reset _only_ the camera settings by deleting
+      the file `/root/.config/mavlink-camera-manager/settings.json` via
+      [the file browser](#file-browser) or [the terminal](#terminal)
+{% end %}
```
Comment (patrickelectric) [inline advanced-usage/index.md:175]: - Or starting the camera manager inside the tmux session with `--reset`

File: advanced-usage/index.md [hunk 8]
```diff
@@ -0,0 +1,372 @@
...
+### System Information
+The system information page provides useful information about the processes,
+network configuration, and computer system BlueOS is running on. It can be
+useful for troubleshooting, and finding if a particular program is using
+excessive resources.
+{{ easy_image(src="system-info-processes", width=600) }}
+{{ easy_image(src="system-info-monitor", width=600) }}
+{{ easy_image(src="system-info-network", width=600) }}
+{{ easy_image(src="system-info-about", width=600) }}
+
```
Comment (patrickelectric) [inline advanced-usage/index.md:239]: It's missing the kernel widget on pirate mode

File: advanced-usage/index.md [hunk 9]
```diff
@@ -0,0 +1,372 @@
...
+The Available Services page provides developer access to the underlying http
+server interfaces of the services upon which BlueOS is based. Each service is
+listed with 
+- the port it is served at
+- a meaningful name
+- a web-page link using the active 
+[network configuration](../getting-started/#interface-access), and where relevant
+- its API documentation (in a live-testable form)
+- the current API version
+{% end %}
```
Comment (patrickelectric) [inline advanced-usage/index.md:284]: Any service that provides an http server with a [title tag](https://www.w3schools.com/tags/tag_title.asp) will be displayed, documentation can also be parsed if follows swagger/open api spec and available in `/docs` or `/v1.0/ui/`

File: advanced-usage/index.md [hunk 10]
```diff
@@ -0,0 +1,372 @@
...
+> http://blueos.local:7777
+{% end %}
+
+{% pirate() %}
+### NMEA Injector
+- Conveys GPS positions (from an NMEA device) to the vehicle via MAVLink messages
+{% end %}
+{{ easy_image(src="nmea-injector", width=600, class="pirate") }}
+{% pirate() %}
+- Setup requires a socket for the NMEA device to connect to, and a MAVLink ID for
```
Comment (patrickelectric) [inline advanced-usage/index.md:321]: requires an UDP socket

File: advanced-usage/index.md [hunk 11]
```diff
@@ -0,0 +1,372 @@
...
+
+{% pirate() %}
+### Terminal
+The terminal provides
+- A tmux session
+- Direct access into the core BlueOS docker container
+- Ready access to the tmux sessions of the core services (`CTRL+b s`)
+   - Useful for seeing logs as they update live
+   - Can kill services if necessary
+- Access to the underlying device via the `red-pill` utility
```
Comment (patrickelectric) [inline advanced-usage/index.md:334]: - To exit from the board terminal and get back to docker, you can use the `exit` command or `CTRL+d`

**Review by patrickelectric — APPROVED** [review]
Looks great, I just added a couple of comments. But we can merge it now and fix it in a next PR.

File: advanced-usage/index.md (hunk 3 above)
Comment (ES-Alexander) [reply advanced-usage/index.md:121]: #new-in-1.1
This branch/PR is specifically for 1.0 - 1.1 changes will be reflected in the 1.1 docs. Thanks for pointing them out though - will make writing 1.1 docs easier :-)

File: advanced-usage/index.md (hunk 4 above)
Comment (ES-Alexander) [reply advanced-usage/index.md:128]: #new-in-1.1

File: advanced-usage/index.md (hunk 5 above)
Comment (ES-Alexander) [reply advanced-usage/index.md:144]: #new-in-1.1

File: advanced-usage/index.md [hunk 12]
```diff
@@ -0,0 +1,372 @@
...
+- Flash firmware onto a connected compatible flight controller board
+
+{{ easy_image(src="firmware", width=600) }}
+
+### Log Browser
+- Built in version of 
+[UAV LogViewer](https://ardupilot.org/copter/docs/common-uavlogviewer.html)
+for powerful analysis of vehicle telemetry
+- Currently only set up to fetch logs automatically from Linux-based autopilots
+   - e.g. Pixhawk **not** yet supported
```
[the remark this answers is not in the document]
Comment (ES-Alexander) [reply advanced-usage/index.md:161]: #new-in-1.1

File: advanced-usage/index.md [hunk 13]
```diff
@@ -0,0 +1,372 @@
...
+      the file `/root/.config/mavlink-camera-manager/settings.json` via
+      [the file browser](#file-browser) or [the terminal](#terminal)
+{% end %}
+
+- After the initial startup, settings are saved and persistent across reboots
+   - Further changes require manually re-configuring streams
+   - New streams need to be manually added
+      - The stream endpoint should be set to `udp://<surface-IP>:<port>`
+      - e.g. `udp://192.168.2.1:5601`
+- The streams are also presented via MAVLink, so QGroundControl (v4.1.7) can
```
[the remark this answers is not in the document]
Comment (ES-Alexander) [reply advanced-usage/index.md:182]: Are you sure? I thought at least TCP was new in 1.1

**Pull request merged**